defold-rive/pluginsrc/com/defold/bob/pipeline/Rive.java

```diff
@@ -192,6 +193,186 @@ protected static StateMachine RIVE_GetStateMachine(RivePointer rive, int index)
     public static native int RIVE_GetVertexCount(RivePointer rive);
     public static native void RIVE_GetVertices(RivePointer rive, Buffer buffer, int bufferSize); // buffer size must be at least VertexSize * VertexCount bytes long
 
+
+    ////////////////////////////////////////////////////////////////////////////////
+    // Render data
+
+    // Matching the struct in vertices.h
+    static public class RiveVertex extends Structure {
+        public float x, y;
+        protected List getFieldOrder() {
+            return Arrays.asList(new String[] {"x", "y"});
+        }
+    }
+
+    // The enums come straight from https://github.com/defold/defold/blob/dev/engine/graphics/src/dmsdk/graphics/graphics.h
+    public enum CompareFunc
+    {
+        COMPARE_FUNC_NEVER    (0),
+        COMPARE_FUNC_LESS     (1),
+        COMPARE_FUNC_LEQUAL   (2),
+        COMPARE_FUNC_GREATER  (3),
+        COMPARE_FUNC_GEQUAL   (4),
+        COMPARE_FUNC_EQUAL    (5),
+        COMPARE_FUNC_NOTEQUAL (6),
+        COMPARE_FUNC_ALWAYS   (7);
+
+        private final int value;
+        private CompareFunc(int v) { this.value = v; }
+        public int getValue() { return this.value; }
+    };
+
+    public enum FaceWinding
+    {
+        FACE_WINDING_CCW (0),
+        FACE_WINDING_CW  (1);
+
+        private final int value;
+        private FaceWinding(int v) { this.value = v; }
+        public int getValue() { return this.value; }
+    };
+
+    public enum StencilOp
+    {
+        STENCIL_OP_KEEP      (0),
+        STENCIL_OP_ZERO      (1),
+        STENCIL_OP_REPLACE   (2),
+        STENCIL_OP_INCR      (3),
+        STENCIL_OP_INCR_WRAP (4),
+        STENCIL_OP_DECR      (5),
+        STENCIL_OP_DECR_WRAP (6),
+        STENCIL_OP_INVERT    (7);
+
+        private final int value;
+        private StencilOp(int v) { this.value = v; }
+        public int getValue() { return this.value; }
+    };
+
+    static public class StencilTestFunc extends Structure {
+        public int m_Func;      // dmGraphics::CompareFunc
+        public int m_OpSFail;   // dmGraphics::StencilOp
+        public int m_OpDPFail;  // dmGraphics::StencilOp
+        public int m_OpDPPass;  // dmGraphics::StencilOp
+        protected List getFieldOrder() {
+            return Arrays.asList(new String[] {"m_Func", "m_OpSFail", "m_OpDPFail", "m_OpDPPass"});
+        }
```

Comment:
I'm still not sure why I need to do this, when all I want is to use the order I've specified the fields in.

(I had one such issue, spending 2h+ tracking down a case where I had switched places of two variables, and not updated this names list)